server/routes/user.js

```diff
@@ -85,7 +86,60 @@ router.get("/list", function (req, res, next) {
       password: config.fhir.password
     }
   }).then(response => {
-    res.status(201).json(response.data);
+    //console.log(response.data);
+    //check if the user
+    var userEntry=[];
```

Comment:
This is certainly one approach to solving this problem. Is there a way to let fhir handle the workload? IE, is it possible to update the initial request to FHIR so that only Persons with a username field will be returned? I believe there is a property called exists that might work.

Reply:
There is search options: _content which is based on the value and _filter with only binary operator such as eq,ne,... which are not usefull for this case. I did not find the exists property.

Reply:
I did some searching on this and talked to Luke briefly. We should be able to use a search parameter for this. Here is an example of one:  http://hl7.org/fhir/searchparameter-example-extension.json.html

We have one for searching by username. We should be able to either adapt that one or do something similar to also search for one that has a username https://github.com/iHRIS/iHRIS/blob/master/resources/SearchParameter/person-username.SearchParameter.json

Reply:
This isn't resolved.

Reply:
I have found the right synthax for _filter